### PR TITLE
Add new error for practice evaluate

### PR DESCRIPTION
**Pull request by vlasy**
## Description

Add new error type and propose new evaluate return type

## Motivation and Context

We want to detect problems during the evaluation of two kinds. 
- some dependency or crucial component is missing and evaluate cannot work without it (we use internal_error for this)
- practice evaluation may result in an ambiguous state (we use practice_evaluation_error for this)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/practices/Java/JavaGitignoreCorrectlySetPractice.ts [hunk 1]
```diff
@@ -50,7 +51,7 @@ export class JavaGitignoreCorrectlySetPractice implements IPractice {
         return PracticeEvaluationResult.practicing;
       }
     }
-    return PracticeEvaluationResult.unknown;
+    throw ErrorFactory.newPracticeEvaluateError('Gitignore is partially correct');
```
Comment (prokopsimek) [inline src/practices/Java/JavaGitignoreCorrectlySetPractice.ts:54]: The error at the beginning of this function is correct, but here should be `notPracticing`?

Maybe the error should be thrown only in `else` block if no package manager has been found.

**Review by adelkahomolova — COMMENTED** [review]
Approve after resolving Prokop's comment

File: src/practices/Java/JavaGitignoreCorrectlySetPractice.ts (hunk 1 above)
Comment (vlasy) [reply src/practices/Java/JavaGitignoreCorrectlySetPractice.ts]: Resolved in 4a5eb4be0a7bf8f4f98c53f6cff212ac9ca13529

**Pull request merged**